test/e2e/config_test.go

```diff
@@ -57,6 +57,8 @@ func TestClusterMonitoringOperatorConfiguration(t *testing.T) {
 	t.Log("asserting that CMO goes degraded after an invalid configuration is pushed")
 	f.AssertOperatorCondition(configv1.OperatorDegraded, configv1.ConditionTrue)(t)
 	f.AssertOperatorCondition(configv1.OperatorAvailable, configv1.ConditionFalse)(t)
```

Comment:
should we be setting `Available=false` in such invalid config cases?

Reply:
I think so yes. Iirc this was discussed previously and the reasoning was that since the operator doesn't sync any components if the config is invalid, we should consider the operator as unavailable.

Reply:
The PR that added that https://github.com/openshift/cluster-monitoring-operator/pull/731 only mentioned degraded (which seems enough and appropriate for me), but no one has complained about it, let's keep it. Thanks for the context.

Reply:
If memory serves, the operator didn't have granular handling of conditions at the time so it was either "Available=False, Degraded=True" or "Available=True, Degraded=False".

Reply:
I could argue that an invalid config should be "Available=True,Degraded=True" reading https://github.com/openshift/enhancements/blob/master/dev-guide/cluster-version-operator/dev/clusteroperator.md#conditions since all operands may be functional (if they were previously functional). But it's also fine to keep it as-is for me.